testing/endtoend/components/beacon_node.go

```diff
@@ -183,29 +175,10 @@ func (node *BeaconNode) generateGenesis(ctx context.Context) (state.BeaconState,
 		return nil, errors.New("Cannot construct bellatrix block, e2e.TestParams.Eth1GenesisBlock == nil")
 	}
 	gb := e2e.TestParams.Eth1GenesisBlock
-	// so the DepositRoot in the BeaconState should be set to the HTR of an empty deposit trie.
-	t, err := trie.NewTrie(params.BeaconConfig().DepositContractTreeDepth)
-	if err != nil {
-		return nil, err
-	}
-	dr, err := t.HashTreeRoot()
-	if err != nil {
-		return nil, err
-	}
-	e1d := &ethpb.Eth1Data{
-		DepositRoot:  dr[:],
-		DepositCount: 0,
-		BlockHash:    gb.Hash().Bytes(),
-	}
-	v := e2etypes.GenesisFork()
-	switch v {
-	case version.Bellatrix:
-		return generateGenesisBellatrix(ctx, gb, e1d)
-	case version.Phase0:
-		return generateGenesisPhase0(ctx, e1d)
-	default:
-		return nil, fmt.Errorf("unsupported genesis fork version %s", version.String(v))
-	}
+	t := e2e.TestParams.CLGenesisTime
+	nvals := params.BeaconConfig().MinGenesisActiveValidatorCount
+	version := e2etypes.GenesisFork()
+	return spectypes.NewPreminedGenesis(ctx, t, nvals, version, gb)
```

Comment:
`NewPreminedGenesis` does most of the heavy lifting now so this component gets much simpler.